microsoft/coe-starter-kit
# ALM Accelerator For Advanced Makers 1.0.20210503.1

Public preview release of the ALM Accelerator For Advanced Makers. Included in this release are the following enhancements/fixes.

The major update in this release is a change to the pipelines to remove the reliance on build completion triggers to simplify setup and provide a better experience when applying branch policies. In the previous release we had a build pipeline and seperate deployment pipelines for each environment. This made setup cumbersome and also didn't address several issues including:

- Ensuring that both the build and deployment to validation succeeded when running the validation build via the branch policy. Instead the previous version would only validate the build and then a user would have to go to the deployment pipeline to determine if the deployment succeeded.
- Allow for deployment without build. In the previous version a build was kicked off for every deployment. The feedback we've gotten is that in some cases (e.g. when deploying to prod) organizations want to be able to deploy the most recent build that was deployed to test.

Both of the above issues have been handled in the latest release pipelines and the setup instructions have been updated to demonstrate how to handle both scenarios by using a single pipeline per environment to which a solution is being deployed.

NOTE: If you are upgrading to the latest public preview you will need to perform the following steps.
- Delete or deactivate your build pipeline.
- Update your Branch policy to point to the pipeline deploy-validation-MySolution
- Upgrade to the latest AA4AM Solution
- Update your pipeline templates repo with the latest from https://github.com/microsoft/coe-alm-accelerator-templates
- Follow the instructions in the SETUPGUIDE.md to create / update your solution deployment pipelines (e.g. deploy-validation-MySolution, deploy-test-MySolution and deploy-prod-MySolution) found here https://github.com/microsoft/coe-starter-kit/blob/main/PowerPlatformDevOpsALM/SETUPGUIDE.md#setting-deployment-pipeline-variables
 

Issues addressed in this release:
1. Pipelines: Update CoE Starter Kit Pipelines to Deploy without Build to Prod #238
2. Pipelines: Non-namespaced Build Task #219
3. AA4AM Docs: Updates #185
4. ALM Accelerator Sample Solution: No longer operable #228
5. AA4AM: Readme Doc #213
6. Service connection related permission issue while running the pipelines #177
7. Pipelines: Update build pipeline for PRs to not use build completion triggers #160
8. Add required connectors to documentation #191
9. AA4AM App: Add hover tooltip for long solution names #112 
10. AA4AM App: Sort Solutions Alphabetically #116
11. Pipelines: Spaces in projects / repos cause errors in pipelines #180
12. Improve readability of GETTINGSTARTED.md #182
13. AA4AM App: Update Custom Connector Description #167
